peps/pep-0747.rst

```diff
@@ -662,12 +666,32 @@ not propose those changes now:
 
    -  Returns annotation expressions
 
+The following signatures accepting union type expressions continue
+to use ``UnionType``:
+
+-  ``builtins.isinstance``
```

Comment:
Type checkers already need to do significant special-casing for `isinstance` and `issubclass` calls. Maybe we just change these to accept `TypeExpr` rather than (or in addition to) `UnionType` and leave it up to the custom type-checker logic to detect and report cases that the runtime implementation of `isinstance` and `issubclass` are not able to handle. That would eliminate the need for `TypeExpr` to be a subtype of `UnionType` (as proposed above, which I think is very problematic).

Reply:
> Maybe we just change these to accept TypeExpr rather than (or in addition to) UnionType ...

Sounds good to me. I think Jelle made a similar suggestion elsewhere in the comments for this PR.